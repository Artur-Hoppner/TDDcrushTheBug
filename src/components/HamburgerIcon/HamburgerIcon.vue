<template>
  <div id="hamburgerIcon">
    <HamburgerMenu>
      <ul class="sidebar-panel-nav"><!-- v-if="this.$route.path !== '/landing'" -->
      <li @click="changeThisToggle"><router-link to="/landing" exact-active-class="active">Home</router-link></li>
      <li @click="changeThisToggle"><router-link to="/shopping" exact-active-class="active">Our Shop</router-link></li>
      <li @click="changeThisToggle"><router-link to="/cart" exact-active-class="active">Cart</router-link></li>
      <li @click="changeThisToggle"><router-link to="/order" exact-active-class="active">Order</router-link></li>
      </ul>
    </HamburgerMenu>
    <nav class="burger-button">
      <img
        class="hamburgerImage"
        src="@/assets/navicon.svg"
        alt="hamburgerImage"
        @click="changeThisToggle"
      />
    </nav>
  </div>
</template>

<script>
import HamburgerMenu from '@/components/HamburgerMenu/HamburgerMenu.vue';

import { mapActions } from 'vuex';
export default {
  name: 'HamburgerIcon',
  components: { HamburgerMenu },
  methods: {
    ...mapActions('hamburgerIcon', ['changeThisToggle']),
    goTo(path) {
      this.$router
        .push(path)
        .then(this.changeThisToggle())
        .catch(err => {
          console.log(err);
        });
    }
  }
};
</script>
<style lang="scss" scoped>
ul.sidebar-panel-nav {
  list-style-type: none;
    display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

ul.sidebar-panel-nav > li > a {
  
  text-decoration: none;
  font-size: 3rem;
    display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding-bottom: 1.5em;

}

ul > li > a {
  color: #fff;
}
.active {
  color: red;
}
</style>