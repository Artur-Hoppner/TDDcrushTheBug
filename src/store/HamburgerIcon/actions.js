const actions = {
  changeThisToggle(context) {
    context.commit('toggle');
  }
}

export default actions;