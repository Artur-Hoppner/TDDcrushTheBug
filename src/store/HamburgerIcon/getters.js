const getters = {
  showHamburger: state => state.isBurgerActive
};

export default getters;
