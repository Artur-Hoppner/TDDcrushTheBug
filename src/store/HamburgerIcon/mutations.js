const mutations = {
  toggle(state) {
    state.isBurgerActive = !state.isBurgerActive
  }
}

export default mutations;