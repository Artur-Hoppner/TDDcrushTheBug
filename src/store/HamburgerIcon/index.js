import mutations from './mutations.js';
import getters from './getters.js';
import actions from './actions.js';

const state = {
  namespaced: true,

  state: {
    isBurgerActive: false
  },
  mutations: mutations,
  getters: getters,
  actions: actions
};

export default state;
