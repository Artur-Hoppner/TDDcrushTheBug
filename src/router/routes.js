import Landing from '../views/Landing/Landing.vue';
import Cart from '../views/Cart/Cart.vue';
import Order from '../views/Order/Order.vue';
import Shopping from '../views/Shopping/Shopping.vue';
import store from '@/store/index.js';

const routes = [
  {
    path: '/landing',
    name: 'Landing',
    component: Landing
  },
  {
    path: '/cart',
    name: 'Cart',
    component: Cart
  },
  {
    path: '/order',
    name: 'Order',
    component: Order,
    beforeEnter(to, from, next) {
      // check if vuex store orderInfo is empty. Cant go to order page without checking out//
      if (store.state.orderInfo !== '') {
        next();
      } else {
        next({
          name: 'Shopping' // back to safety route //
        });
      }
    }
  },
  {
    path: '/shopping',
    name: 'Shopping',
    component: Shopping
  }
];

export default routes;
