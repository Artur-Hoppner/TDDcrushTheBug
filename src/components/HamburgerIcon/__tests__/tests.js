import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import hamburgerIcon from '../../../store/HamburgerIcon/index.js';
import HamburgerIcon from '@/components/HamburgerIcon/HamburgerIcon.vue';
import VueRouter from "vue-router"
import routes from '../../../router/routes.js'

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('User clicks icon/button', () => {
  let modules, store, router;

  modules = {
    hamburgerIcon
  };


  beforeEach(() => {
    //mock our store
    store = new Vuex.Store({
      modules
    });
    router = new VueRouter({
      routes
    });
  });

  test('Test so that everything renders correctly', () => {
    const wrapper = shallowMount(HamburgerIcon, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('Icon should show when rendered', async () => {
    const wrapper = shallowMount(HamburgerIcon, {
      router,
      store,
      localVue,
    });

    const buttonIcon = wrapper.find('.hamburgerImage');

    expect(buttonIcon.exists()).toBe(true);
  });

  test('Icon should toggle menu on click', async () => {
    const hamburgerIcon = {
      namespaced: true,
      state: {},
      mutations: {},
      actions: {
        changeThisToggle: jest.fn()
      },
      getters: {
        showHamburger() {
          return null;
        }
      }
    };
    const store = new Vuex.Store({
      modules: {
        hamburgerIcon
      }
    });
    const wrapper = mount(HamburgerIcon, {
      router,
      store,
      localVue,
    });

    const buttonIcon = wrapper.find('.hamburgerImage');

    await buttonIcon.trigger('click');

    expect(hamburgerIcon.actions.changeThisToggle).toHaveBeenCalled();
  });
});
