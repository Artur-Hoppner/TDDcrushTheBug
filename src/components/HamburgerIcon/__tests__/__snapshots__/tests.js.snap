// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User clicks icon/button Test so that everything renders correctly 1`] = `
<div
  id="hamburgerIcon"
>
  <hamburgermenu-stub>
    <ul
      class="sidebar-panel-nav"
    >
      <li>
        <router-link-stub
          ariacurrentvalue="page"
          event="click"
          exactactiveclass="active"
          tag="a"
          to="/landing"
        >
          Home
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          ariacurrentvalue="page"
          event="click"
          exactactiveclass="active"
          tag="a"
          to="/shopping"
        >
          Our Shop
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          ariacurrentvalue="page"
          event="click"
          exactactiveclass="active"
          tag="a"
          to="/cart"
        >
          Cart
        </router-link-stub>
      </li>
       
      <li>
        <router-link-stub
          ariacurrentvalue="page"
          event="click"
          exactactiveclass="active"
          tag="a"
          to="/order"
        >
          Order
        </router-link-stub>
      </li>
    </ul>
  </hamburgermenu-stub>
   
  <nav
    class="burger-button"
  >
    <img
      alt="hamburgerImage"
      class="hamburgerImage"
      src="@/assets/navicon.svg"
    />
     
    <hamburgermenu-stub />
  </nav>
</div>
`;
